File: bbb-voice/src/main/java/org/bigbluebutton/voiceconf/messaging/messages/UserDisconnectedFromGlobalAudio.java
package org.bigbluebutton.voiceconf.messaging.messages;

import java.util.HashMap;

import org.bigbluebutton.voiceconf.messaging.Constants;
import org.bigbluebutton.voiceconf.messaging.MessageBuilder;


public class UserDisconnectedFromGlobalAudio {
	public static final String NAME = "UserDisconnectedFromGlobalAudioMsg";

	public final String voiceConf;
	public final String name;
	public final String userId;
	
	public UserDisconnectedFromGlobalAudio(String voiceConf, String userId, String name) {
		this.voiceConf = voiceConf;
		this.userId = userId;
		this.name = name;
	}
	
	public String toJson() {
		HashMap<String, Object> body = new HashMap<String, Object>();
		body.put(Constants.USER_ID, userId);
		body.put(Constants.NAME, name);

		return MessageBuilder.buildJson(NAME, voiceConf, body);
	}
}

File: labs/bbb-webrtc-sfu/lib/bbb/messages/audio/UserDisconnectedFromGlobalAudio.js
var inherits = require('inherits');
var OutMessage = require('../OutMessage');

module.exports = function(Constants) {
  function UserDisconnectedFromGlobalAudio(voiceConf, userId, name) {
    UserDisconnectedFromGlobalAudio.super_.call(this, Constants.GLOBAL_AUDIO_DISCONNECTED);

    this.payload = {};
    this.payload[Constants.VOICE_CONF] = voiceConf;
    this.payload[Constants.USERID] = userId;
    this.payload[Constants.NAME] = name;
  };

  inherits(UserDisconnectedFromGlobalAudio, OutMessage);
  return UserDisconnectedFromGlobalAudio;
}
